### DISCUSSION TITLE
OnnxTR & GPU

### DISCUSSION THREAD

**Discussion by mykolamelnykml** [mindee/doctr#1659] [General]
How to enable to use GPU for OnnxTR?

**Comment by felixdittrich92** [comment]
Hi @kolia1985 :wave:;

Would be better to ask Onnx related things directly in https://github.com/felixdittrich92/OnnxTR/discussions :)

- first: You need to install [CUDA](https://onnxruntime.ai/docs/build/eps.html#cuda) be careful from my experience this needs to match exactly with the versions provided in [Versions](https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#requirements) and your host machine (nvidia-smi / nvcc --version) - Unfortunately this part is a bit tricky (by onnxruntime)
- sec: Then you can install `pip install "onnxtr[gpu]"` and it should auto detect your GPU and prefer it

**Reply by mykolamelnykml** [reply]
Thank you
